Data Structures/Queues/RingQueue.java

```diff
@@ -0,0 +1,94 @@
+public interface Queue {
+
+	public void add(Object object);
+
+	public Object first();
+
+	public Object remove();
+
+	public int size();
+
+}
+public class RingQueue implements Queue {
+	private Object[] objArray = null;
+	private int size;
+	private int front;
+	private int rear;
+
+	public RingQueue(int n) {
+		// Create an array of size n
+		objArray = new Object[n];
+		size = 0;
+		front = 0;
```

Comment:
The problem is that for ```(front+1) % objArray.length``` the pointer ```front``` never reaches the element with index 0 of the array. I would ```front``` set equal to ```-1```.